In package-manager-install mode, npm install could fail as node is not added to the PATH when running

```
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] Running Quinoa package manager install command: /workspace/source/node/node /workspace/source/node/node_modules/npm/bin/npm-cli.js install
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm notice 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm notice New minor version of npm available! 8.15.0 -> 8.19.2
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.2>
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm notice Run `npm install -g npm@8.19.2` to update!
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm notice 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! code 127
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! path /workspace/source/src/main/webui/node_modules/esbuild
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! command failed
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! command sh /tmp/postinstall-fe388bfb.sh
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! /tmp/postinstall-fe388bfb.sh: line 1: node: command not found
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! A complete log of this run can be found in:
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!     /root/.npm/_logs/2022-09-20T15_56_41_337Z-debug-0.log
```

This needs to be fixed by adding node to the PATH env.

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java
@@ -13,6 +13,7 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.AbstractMap;
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
@@ -28,6 +29,7 @@ import org.jboss.logging.Logger;
 import io.quarkiverse.quinoa.QuinoaHandlerConfig;
 import io.quarkiverse.quinoa.QuinoaRecorder;
 import io.quarkiverse.quinoa.deployment.packagemanager.PackageManager;
+import io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerInstall;
 import io.quarkus.arc.deployment.BeanContainerBuildItem;
 import io.quarkus.builder.BuildException;
 import io.quarkus.deployment.IsNormal;
@@ -92,12 +94,15 @@ public class QuinoaProcessor {
                     "No package.json found in Web UI directory: '" + configuredDir + "'");
         }
         Optional<String> packageManagerBinary = quinoaConfig.packageManager;
+        List<String> paths = new ArrayList<>();
         if (quinoaConfig.packageManagerInstall.enabled) {
-            final String result = install(quinoaConfig.packageManagerInstall, uiDirEntry.getValue());
-            packageManagerBinary = Optional.of(result);
+            final PackageManagerInstall.Installation result = install(quinoaConfig.packageManagerInstall,
+                    uiDirEntry.getValue());
+            packageManagerBinary = Optional.of(result.getPackageManagerBinary());
+            paths.add(result.getNodePath());
         }
         PackageManager packageManager = autoDetectPackageManager(packageManagerBinary,
-                quinoaConfig.packageManagerCommand, uiDirEntry.getKey());
+                quinoaConfig.packageManagerCommand, uiDirEntry.getKey(), paths);
         final boolean alreadyInstalled = Files.isDirectory(packageManager.getDirectory().resolve("node_modules"));
         final boolean packageFileModified = liveReload.isLiveReload()
                 && liveReload.getChangedResources().stream().anyMatch(r -> r.equals(packageFile.toString()));

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/EffectiveCommands.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/EffectiveCommands.java
@@ -1,18 +1,23 @@
 package io.quarkiverse.quinoa.deployment.packagemanager;
 
+import java.io.File;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 import io.quarkus.runtime.LaunchMode;
 
 class EffectiveCommands implements PackageManagerCommands {
+    private static final String PATH_ENV_VAR = "PATH";
     private final PackageManagerCommands defaultCommands;
     private final PackageManagerCommandConfig commandsConfig;
+    private final List<String> paths;
 
-    EffectiveCommands(PackageManagerCommands defaultCommands, PackageManagerCommandConfig commandsConfig) {
+    EffectiveCommands(PackageManagerCommands defaultCommands, PackageManagerCommandConfig commandsConfig, List<String> paths) {
         this.defaultCommands = defaultCommands;
         this.commandsConfig = commandsConfig;
+        this.paths = paths;
     }
 
     @Override
@@ -61,9 +66,39 @@ class EffectiveCommands implements PackageManagerCommands {
                 .map(a -> commandsConfig.prependBinary ? defaultCommands.binary() + " " + a : a);
     }
 
-    private Map<String, String> environment(Command c, Map<String, String> override) {
-        Map<String, String> environment = new HashMap<>(c.envs);
-        environment.putAll(override);
+    private Map<String, String> environment(final Command c, final Map<String, String> additionalEnvironment) {
+        final Map<String, String> environment = new HashMap<>(System.getenv());
+
+        environment.putAll(c.envs);
+
+        if (additionalEnvironment != null) {
+            environment.putAll(additionalEnvironment);
+        }
+
+        if (PackageManager.isWindows()) {
+            for (final Map.Entry<String, String> entry : environment.entrySet()) {
+                final String pathName = entry.getKey();
+                if (PATH_ENV_VAR.equalsIgnoreCase(pathName)) {
+                    final String pathValue = entry.getValue();
+                    environment.put(pathName, extendPathVariable(pathValue, paths));
+                }
+            }
+        } else {
+            final String pathValue = environment.get(PATH_ENV_VAR);
+            environment.put(PATH_ENV_VAR, extendPathVariable(pathValue, paths));
+        }
+
         return environment;
+    }
+
+    private String extendPathVariable(final String existingValue, final List<String> paths) {
+        final StringBuilder pathBuilder = new StringBuilder();
+        for (final String path : paths) {
+            pathBuilder.append(path).append(File.pathSeparator);
+        }
+        if (existingValue != null) {
+            pathBuilder.append(existingValue).append(File.pathSeparator);
+        }
+        return pathBuilder.toString();
     }
 }

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManager.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManager.java
@@ -12,6 +12,7 @@ import java.net.SocketTimeoutException;
 import java.net.URL;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
@@ -126,7 +127,7 @@ public class PackageManager {
     }
 
     public static PackageManager autoDetectPackageManager(Optional<String> binary,
-            PackageManagerCommandConfig packageManagerCommands, Path directory) {
+            PackageManagerCommandConfig packageManagerCommands, Path directory, List<String> paths) {
         String resolved = null;
         if (binary.isEmpty()) {
             if (Files.isRegularFile(directory.resolve("yarn.lock"))) {
@@ -142,22 +143,23 @@ public class PackageManager {
         } else {
             resolved = binary.get();
         }
-        return new PackageManager(directory, resolveCommands(resolved, packageManagerCommands));
+        return new PackageManager(directory, resolveCommands(resolved, packageManagerCommands, paths));
     }
 
     public static boolean isWindows() {
         return IS_WINDOWS;
     }
 
-    static PackageManagerCommands resolveCommands(String binary, PackageManagerCommandConfig packageManagerCommands) {
+    static PackageManagerCommands resolveCommands(String binary, PackageManagerCommandConfig packageManagerCommands,
+            List<String> paths) {
         if (binary.contains(PNPMPackageManagerCommands.pnpm)) {
-            return new EffectiveCommands(new PNPMPackageManagerCommands(binary), packageManagerCommands);
+            return new EffectiveCommands(new PNPMPackageManagerCommands(binary), packageManagerCommands, paths);
         }
         if (binary.contains(NPMPackageManagerCommands.npm)) {
-            return new EffectiveCommands(new NPMPackageManagerCommands(binary), packageManagerCommands);
+            return new EffectiveCommands(new NPMPackageManagerCommands(binary), packageManagerCommands, paths);
         }
         if (binary.contains(YarnPackageManagerCommands.yarn)) {
-            return new EffectiveCommands(new YarnPackageManagerCommands(binary), packageManagerCommands);
+            return new EffectiveCommands(new YarnPackageManagerCommands(binary), packageManagerCommands, paths);
         }
         throw new UnsupportedOperationException("Unsupported package manager binary: " + binary);
     }

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java
@@ -13,15 +13,15 @@ import com.github.eirslett.maven.plugins.frontend.lib.ProxyConfig;
 import io.quarkus.runtime.configuration.ConfigurationException;
 
 public final class PackageManagerInstall {
-    private static final String INSTALL_PATH = "node";
-    private static final String NODE_PATH = INSTALL_PATH + "/node";
-    private static final String NPM_PATH = INSTALL_PATH + "/node_modules/npm/bin/npm-cli.js";
+    private static final String INSTALL_SUB_PATH = "node";
+    private static final String NODE_BINARY = PackageManager.isWindows() ? "node.exe" : "node";
+    private static final String NPM_PATH = INSTALL_SUB_PATH + "/node_modules/npm/bin/npm-cli.js";
 
     private PackageManagerInstall() {
 
     }
 
-    public static String install(PackageManagerInstallConfig config, final Path projectDirectory) {
+    public static Installation install(PackageManagerInstallConfig config, final Path projectDirectory) {
         final Path installDirPath = projectDirectory.resolve(config.installDir.orElse(""));
         final File installDirFile = installDirPath.toFile();
         FrontendPluginFactory factory = new FrontendPluginFactory(installDirFile, installDirFile);
@@ -53,18 +53,36 @@ public final class PackageManagerInstall {
         }
     }
 
-    private static String resolveInstalledNpmBinary(Path installDirectory) {
-        final Path nodePath = installDirectory.resolve(NODE_PATH)
+    private static Installation resolveInstalledNpmBinary(Path installDirectory) {
+        final Path nodeBinPath = installDirectory.resolve(INSTALL_SUB_PATH)
                 .toAbsolutePath();
         final Path npmPath = installDirectory.resolve(NPM_PATH).toAbsolutePath();
-        if (PackageManager.isWindows()) {
-            return convertToWindowsPath(nodePath + ".exe " + npmPath);
-        }
-        return nodePath + " " + npmPath;
+        final String platformNodeBinPath = convertToWindowsPathIfNeeded(nodeBinPath.toString());
+        final String platformNPMPath = convertToWindowsPathIfNeeded(npmPath.toString());
+        final String packageManagerBinary = NODE_BINARY + " " + platformNPMPath;
+        return new Installation(platformNodeBinPath, packageManagerBinary);
     }
 
-    public static String convertToWindowsPath(String path) {
-        return path.replaceAll("/", "\\\\");
+    public static String convertToWindowsPathIfNeeded(String path) {
+        return PackageManager.isWindows() ? path.replaceAll("/", "\\\\") : path;
+    }
+
+    public static class Installation {
+        private final String nodePath;
+        private final String packageManagerBinary;
+
+        public Installation(String nodePath, String packageManagerBinary) {
+            this.nodePath = nodePath;
+            this.packageManagerBinary = packageManagerBinary;
+        }
+
+        public String getNodePath() {
+            return nodePath;
+        }
+
+        public String getPackageManagerBinary() {
+            return packageManagerBinary;
+        }
     }
 
 }

--- a/deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java
+++ b/deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java
@@ -1,6 +1,5 @@
 package io.quarkiverse.quinoa.test;
 
-import static io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerInstall.convertToWindowsPath;
 import static io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest.getWebUITestDirPath;
 import static io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest.isWindows;
 import static org.assertj.core.api.Assertions.assertThat;
@@ -10,6 +9,7 @@ import java.nio.file.Path;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
+import io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerInstall;
 import io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest;
 import io.quarkus.test.QuarkusUnitTest;
 
@@ -39,7 +39,7 @@ public class QuinoaPackageManagerInstallPrependBinaryTest {
     }
 
     private static String convertToWindowsPathIfNeeded(String path) {
-        return isWindows() ? convertToWindowsPath(path) : path;
+        return isWindows() ? PackageManagerInstall.convertToWindowsPathIfNeeded(path) : path;
     }
 
 }

--- a/deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallTest.java
+++ b/deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallTest.java
@@ -1,6 +1,5 @@
 package io.quarkiverse.quinoa.test;
 
-import static io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerInstall.convertToWindowsPath;
 import static io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest.getWebUITestDirPath;
 import static io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest.isWindows;
 import static org.assertj.core.api.Assertions.assertThat;
@@ -10,6 +9,7 @@ import java.nio.file.Path;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
+import io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerInstall;
 import io.quarkiverse.quinoa.deployment.testing.QuinoaQuarkusUnitTest;
 import io.quarkus.test.QuarkusUnitTest;
 
@@ -36,7 +36,7 @@ public class QuinoaPackageManagerInstallTest {
     }
 
     private static String convertToWindowsPathIfNeeded(String path) {
-        return isWindows() ? convertToWindowsPath(path) : path;
+        return isWindows() ? PackageManagerInstall.convertToWindowsPathIfNeeded(path) : path;
     }
 
 }

